bubble_sort_by.rb

```diff
@@ -0,0 +1,12 @@
+def bubble_sort_by(arr)
+  arr.length.times do
+    arr.each_with_index do |_x, i|
+      left = arr[i]
+      arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[-1] != left && yield(arr[i], arr[i + 1]).positive?
+    end
+  end
+  arr
+end
+p bubble_sort_by(%w[hi hello hey]) { |left, right|
+  left.length - right.length
+}
```

Comment:
Lines 10, 11 and 12. where you have 
'p bubble_sort_by(%w[hi hello hey]) { |left, right|
  left.length - right.length
}'
 it is a debugging code. Kindly deleted or move to another branch. :wrench: 

![debugging_requirement](https://user-images.githubusercontent.com/64173189/107528223-9933d500-6b87-11eb-8062-fcb5e9d9ed08.png)